tests: net: socket: getaddrinfo: move to new ztest API

hunterhu3000 opened the pull request:
Move getaddrinfo tests to use new ztest API
TEST=twister -T tests/net/socket/getaddrinfo

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/getaddrinfo/src/main.c:
```diff
@@ -212,9 +212,11 @@ void test_getaddrinfo_setup(void)
 }
 
-void test_getaddrinfo_ok(void)
+ZTEST(socket_getaddrinfo_test_suite, test_getaddrinfo_ok)
```
no need for _test_suite here.

hunterhu3000 replied:
> no need for _test_suite here.

Updated. Thanks to review.

hunterhu3000 committed:
tests: net: socket: getaddrinfo: move to new ztest API

Move getaddrinfo tests to use new ztest API
TEST=twister -T tests/net/socket/getaddrinfo

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case cannot be run twice.
